src/main/java/org/broadinstitute/hellbender/utils/smithwaterman/SmithWatermanJavaAligner.java

```diff
@@ -48,6 +50,10 @@ public static SmithWatermanJavaAligner getInstance() {
      */
     private SmithWatermanJavaAligner(){}
 
+    public SmithWatermanJavaAligner(boolean haplotypeToref){
```

Comment:
Hmm... come to think of it, i think the safest thing to do with this code is to make an interface for `SmithWatermanJavaAligner` that contains all of the common code and then two subclasses, one being the current implementation and the other being your optimized version. Then you can add to `SmithWatermanAligner` a new implementation `JAVA_OPTIMIZED` that contains your optimizations. This will help encapsulate your changes and allow this branch to get into the gatk much sooner.